File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robot/actions/IntakeIn.java
package org.firstinspires.ftc.teamcode.robot.actions;

import androidx.annotation.NonNull;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.Action;

import org.firstinspires.ftc.teamcode.robot.Intake;

public class IntakeIn implements Action {
    private boolean init = false;
    private Intake i;

    @Override
    public boolean run(@NonNull TelemetryPacket packet) {
        if(!init) {
            Intake.target = 0;
            init = true;
        }

        return i.iM.getCurrentPosition() > 40;
    }

    public IntakeIn(Intake li) {
        i = li;
    }

    public Action intakeIn(Intake li) {
        return new IntakeIn(li);
    }
}

File: road-runner-quickstart-master/TeamCode/src/main/java/org/firstinspires/ftc/teamcode/dairy/actions/IntakeIn.kt
package org.firstinspires.ftc.teamcode.dairy.actions

import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.Action
import org.firstinspires.ftc.teamcode.dairy.subsystems.Intake
import org.firstinspires.ftc.teamcode.dairy.util.Waiter

class IntakeIn() : Action {
    private var init = false
    private lateinit var waiter: Waiter

    override fun run(p: TelemetryPacket): Boolean {
        if (!init) {
            Intake.target = 0.0
            Intake.pidused = true
            init = true
        }

        if(Intake.intake!!.currentPosition <= 20) {
            Intake.pidused = false
        }

        return Intake.intake!!.currentPosition <= 19
    }

    companion object {
        fun intakeIn(): Action {
            return IntakeIn()
        }}
}